api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForMoreThanXmonthsCalcultion.java

```diff
@@ -62,23 +64,43 @@ public CalculationResultMap evaluate(
     Date oneYearBefore = EptsCalculationUtils.addMonths(onOrBefore, -12);
     CalculationResultMap lastVl =
         ePTSCalculationService.lastObs(
-            Arrays.asList(labEncounterType, adultFollowup, childFollowup),
+            Arrays.asList(labEncounterType, adultFollowup, childFollowup, masterCard),
             viralLoadConcept,
             location,
             oneYearBefore,
             onOrBefore,
             cohort,
             context);
+    CalculationResultMap qViralLoadResultsMap =
+        ePTSCalculationService.lastObs(
+            Arrays.asList(labEncounterType, adultFollowup, childFollowup, masterCard),
+            qualitativeViralLoadResults,
+            location,
+            oneYearBefore,
+            onOrBefore,
+            cohort,
+            context);
 
     for (Integer ptId : cohort) {
       boolean isOnArtForMoreThan3Months = false;
       SimpleResult artStartDateResult = (SimpleResult) arvsInitiationDateMap.get(ptId);
-      Obs lastVlObs = EptsCalculationUtils.resultForPatient(lastVl, ptId);
-      if (checkNotNull(artStartDateResult, lastVlObs)) {
-        Date artStartDate = (Date) artStartDateResult.getValue();
-        Date lastVlDate = lastVlObs.getObsDatetime();
-        if (checkNotNull(artStartDate) && isAtLeastThreeMonthsLater(artStartDate, lastVlDate)) {
-          isOnArtForMoreThan3Months = true;
+      Date artStartDate;
+      if (artStartDateResult != null) {
+        artStartDate = (Date) artStartDateResult.getValue();
+        Obs lastVlObs = EptsCalculationUtils.resultForPatient(lastVl, ptId);
+        Obs lastQualitativeResults =
+            EptsCalculationUtils.resultForPatient(qViralLoadResultsMap, ptId);
+        if (checkNotNull(artStartDateResult, lastVlObs)) {
+
+          Date lastVlDate = lastVlObs.getObsDatetime();
```

Comment:
Think we should also check if the lastVlDate is not null, I believe that will not make much effect  but as we checked on line 96 we should do the same!

Reply:
done